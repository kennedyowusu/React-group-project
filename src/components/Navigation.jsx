import React from 'react';
import { NavLink } from 'react-router-dom';
import planet from '../assets/images/planet.png';

export default function Navigation() {
  const activeState = {
    textDecoration: 'underline',
    color: '#52a3ff',
  };

  const notActiveState = {
    textDecoration: 'none',
    color: '#52a3ff',
  };

  return (
    <nav className="nav-menu">
      <div className="brand">
        <img src={planet} alt="planet-logo" className="app-logo" />
        <h1 className="logo-title">Space Travelers&apos; Hub</h1>
      </div>
      <div>
        <ul className="nav-links">
          <li className="nav-link">
            <NavLink
              to="/"
              style={({ isActive }) => (isActive
                ? activeState
                : {
                  ...notActiveState,
                })}
            >
              Rockets
            </NavLink>
          </li>
          <li className="nav-link">
            <NavLink
              to="/missions"
              style={({ isActive }) => (isActive
                ? activeState
                : {
                  ...notActiveState,
                })}
            >
              Missions
            </NavLink>
          </li>
          <div className="nav-pipe">|</div>
          <li className="nav-link">
            <NavLink
              to="/profile"
              style={({ isActive }) => (isActive
                ? activeState
                : {
                  ...notActiveState,
                })}
            >
              My Profile
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
}
