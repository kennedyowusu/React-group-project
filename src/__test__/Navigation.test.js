import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';

import Navigation from '../components/Navigation';

describe('Test Navigation Component', () => {
  test('navigation component should match with snapshot', () => {
    const navigation = renderer
      .create(
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>,
      )
      .toJSON();

    expect(navigation).toMatchSnapshot();
  });

  test('navigation component should render correctly', () => {
    render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    );

    expect(screen.getByText('Rockets')).toBeInTheDocument();
    expect(screen.getByText('Missions')).toBeInTheDocument();
    expect(screen.getByText('My Profile')).toBeInTheDocument();
  });
});
