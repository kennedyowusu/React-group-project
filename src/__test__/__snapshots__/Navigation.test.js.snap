// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navigation Component navigation component should match with snapshot 1`] = `
<nav
  className="nav-menu"
>
  <div
    className="logo-title"
  >
    <img
      alt="planet-logo"
      className="app-logo"
      src="planet.png"
    />
    <h1>
      Space Travelers Hub
    </h1>
  </div>
  <div>
    <ul
      className="nav-links"
    >
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "blue",
              "textDecoration": "underline",
            }
          }
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className=""
          href="/missions"
          onClick={[Function]}
          style={
            Object {
              "color": "blue",
              "textDecoration": "none",
            }
          }
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className=""
          href="/profile"
          onClick={[Function]}
          style={
            Object {
              "color": "blue",
              "textDecoration": "none",
            }
          }
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
